pandas/core/groupby/groupby.py

```diff
@@ -1861,9 +1863,13 @@ def pre_processor(vals: np.ndarray) -> Tuple[np.ndarray, Optional[Type]]:
                 )
 
             inference = None
-            if is_integer_dtype(vals):
+            if is_integer_dtype(vals.dtype):
```

Comment:
this is fine for here, but I think we need a generic 'convert_this_to_something_we_can_use_in_cython)` method on EA. @jbrockmendel @jorisvandenbossche @TomAugspurger

Reply:
As discussed [here](https://github.com/pandas-dev/pandas/issues/33276#issuecomment-608711488), i think the `EA.convert_this_to_something_we_can_use_in_cython` is going to look something like `_ordinal_values`, at least for methods like quantile that are ordering-based